Patch-Source: https://github.com/llvm/llvm-project/pull/130178
---
From 7a68b5504585dd25ccd0fe189527173e01b82c92 Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Thu, 6 Mar 2025 22:37:41 +0100
Subject: [PATCH] [ORC] skip reoptimization tests on ARM

It failed on armv7 with "Architecture not supported" which is due to
StubTests being not supported on ARM

	/builds/fossdd/aports/main/llvm20/src/llvm-project-20.1.0.src/llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp:140: Failure
	Value of: llvm::detail::TakeError(RM.takeError())
	Expected: succeeded
	  Actual: failed  (Architecture not supported) (of type llvm::detail::ErrorHolder)
---
 llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp b/llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp
index 78dc0f24c0f99..d1afa87e3a043 100644
--- a/llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp
+++ b/llvm/unittests/ExecutionEngine/Orc/ReOptimizeLayerTest.cpp
@@ -55,6 +55,10 @@ class ReOptimizeLayerTest : public testing::Test {
     if (Triple.isPPC())
       GTEST_SKIP();
 
+    // ARM is not supported yet.
+    if (Triple.isARM())
+      GTEST_SKIP();
+
     auto EPC = SelfExecutorProcessControl::Create();
     if (!EPC) {
       consumeError(EPC.takeError());
